### DISCUSSION TITLE
Is it possible to do a 2D meshing with pyFluent?

### DISCUSSION THREAD

**Discussion by NL-5** [ansys/pyfluent#2629] [Q&A] [answered]
I am doing a 2D wind simulation now. And I have searched everywhere on the internet try to find a way to do 2D meshing with pyFluent but couldn't find any. So is it possible at all?

**Comment by antoine-khairallah** [comment, marked as answer]
Hello, unfortunately it is not possible to create a 2D mesh with pyfluent because there is no way of creating 2D mesh in FluentMeshing. 
If you are looking to mesh a geometry with python, I recommend you to use gmsh python library: https://gmsh.info/doc/texinfo/gmsh.html
You can find lots of tutorials and it is possible to export your mesh and then run a simulation using pyFluent in solver mode. 
Hope it helps

**Reply by NL-5** [reply]
Thanks! That really helps me a lot. Sorry if my question is too naive, CFD is very new to me. Can gmsh created mesh be set an inlet and outlet in following Fluent process? I hope to integrate this workflow at least from Ansys geometry file .agdb (or .stl, but it seems that stl doesn't support 2D) to the final wind simulation result.

**Reply by antoine-khairallah** [reply]
Do you really need to work in python ? If not you should use the basic ansys workbench: AnsysMeshing + AnsysFluent and work your case from the software user interface. You have plenty of tutorials available online. 
If you need to work in python you can use the .unv (ideas-universal) file format export in gmsh. You can then import you mesh in fluent and work the boundary conditions (inlet, outlet, axis etc...) from there.

**Reply by NL-5** [reply]
Python is not necessary. In my case I have to generate hundreds of samples, so using python script to save time is the best method I can think of. I start from cad files of buildings layout, then use a rectangle to subtract each wind flow area. I can set named selection in designModeler, but it'd be kinda problemetic if the mesh couldn't preserve the info so pyfluent don't know what to do. Anyway thanks for your advice! I will try gmsh 👍

**Reply by Ericyang978** [reply]
As a follow up question to the author's post, FLUENT Meshing has a built in import 2D Mesh option, is there similar functionality within pyfluent?

**Reply by NL-5** [reply]
Hi yes it is similar. When opening a new session, set the version to 2d will do.

**Reply by peterpodj** [reply]
> Python is not necessary. In my case I have to generate hundreds of samples, so using python script to save time is the best method I can think of. I start from cad files of buildings layout, then use a rectangle to subtract each wind flow area. I can set named selection in designModeler, but it'd be kinda problemetic if the mesh couldn't preserve the info so pyfluent don't know what to do. Anyway thanks for your advice! I will try gmsh 👍

Hi, I am working through a project similar to yours, were you eventually able to get a meshing workflow working that used your python generated geometries? Did you use designmodeler in the end?

**Reply by HugoP707** [reply]
@antoine-khairallah Hello, you mentioned there is a way to export from gmsh to Fluent, do you have any references that explain how its done? Ive been searching all over and cannot find a way to do it.

**Comment by NL-5** [comment]
To those who may be interested, it is also possible to integrate a GUI script for workbench with Python/JavaScript for mesh to automatically generate meshes in Ansys.

**Reply by Ericyang978** [reply]
Thank you! Where can we find more resources for developing a GUI script as you've described?

**Comment by prmukherj** [comment]
It is possible to create a 2d mesh in PyFluent via Fluent now. Please visit the documentation page for more details:

search for "2D meshing workflow" in https://github.com/ansys/pyfluent/blob/main/doc/source/user_guide/meshing/new_meshing_workflows.rst
